src/Compilers/Core/Portable/InternalUtilities/Hash.cs

```diff
@@ -87,7 +87,7 @@ internal static int CombineValues<T>(T[]? values, int maxItemsToHash = int.MaxVa
                 // Should end up with a constrained virtual call to object.GetHashCode (i.e. avoid boxing where possible).
                 if (value != null)
                 {
-                    hashCode = Hash.Combine(value.GetHashCode(), hashCode);
+                    hashCode = Hash.Combine(EqualityComparer<T>.Default.GetHashCode(value), hashCode);
```

Comment:
>EqualityComparer<T>.Default.GetHashCode(value) [](start = 44, length = 46)

What is the reason for this change?

Reply:
> What is the reason for this change?

Yes, please, add a comment explaining the reason so that the change won't be undone unintentionally.

---
In reply to: [403348616](https://github.com/dotnet/roslyn/pull/43060#discussion_r403348616) [](ancestors = 403348616)

Reply:
I'll revert this change. PerfView is showing allocations of NamingStyle from this method even though the new NamingStyle.GetHashCode implementation is called. I'm not sure why, but the savings from this specific change are not great enough to warrant deep investigation.

Reply:
So the claim about a constrained `object.GetHashCode()` is somewhat correct. There are some scenarios where it might not work (e.g. tiered compilation), but the big case is it fails to account for this:

https://github.com/dotnet/runtime/blob/c21c7fd280aed2ba12ed3807c1082e8aba95a602/src/coreclr/src/System.Private.CoreLib/src/System/Collections/Generic/ComparerHelpers.cs#L78-L108